Automattic/wp-calypso
# Pattern Library: Render anchor elements in type and view switchers

Fixes https://github.com/Automattic/dotcom-forge/issues/6561

## Proposed Changes

This PR originally just refactored `client/my-sites/patterns/components/pattern-library/index.tsx` by spinning out `TypeToggle` into a separate component. However, after reviewing #89661, I realized that it would probably be beneficial if Google could index the page layouts separately (since we're looking to add separate descriptions to those pages).

Consequently, I've replicated the style of `ToggleGroupControl` from `@wordpress/components` and implemented `TypeToggle` with `<a>` elements instead.

After some further discussion, I opted to do the same for `ViewToggle`. See https://github.com/Automattic/wp-calypso/pull/89729#issuecomment-2077446092 for more details.

## Testing Instructions



1. Navigate to `/patterns/about`
2. Test the pattern type switcher
3. Ensure that it works the same as before
4. Ensure that a `calypso_pattern_library_type_switch` Tracks event is fired when using the switcher
5. Test the view switcher
6. Ensure that it works the same as before
7. Ensure that a `calypso_pattern_library_view_switch` Tracks event is fired when using the switcher
